File: programmers/올바른 괄호.py
def solution(s):
#     stack = []

#     for i in s:
#         if i == ')' and stack and stack[-1] == '(':
#             stack.pop()
#         else:
#             stack.append(i)


#     return False if stack else True
    pair = 0
    for x in s:
        if pair < 0: break
        pair = pair + 1 if x == "(" else pair - 1 if x == ")" else pair
        print(x, pair)
    return pair == 0

print(solution('(((()))'))

File: swift_algorithm/swift_algorithm/올바른 괄호.swift
import Foundation

func solution(_ s:String) -> Bool
{
    var ans:Bool = true
    
    var stack: [Character] = []
    
    for char in s {
        if char == ")" {
            guard let last = stack.last, last == "(" else {
                ans = !ans
                break
            }
            stack.removeLast()
            
        } else {
            stack.append(char)
        }
    }

    return 0 < stack.count ? false : ans
}
